src/test/java/com/nvs/store/controller/ProductControllerTest.java

```diff
@@ -0,0 +1,132 @@
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static org.hamcrest.Matchers.is;
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.Mockito.*;
+import static org.springframework.security.test.web.servlet.setup.SecurityMockMvcConfigurers.springSecurity;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+public class ProductControllerTest {
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Autowired
+    private WebApplicationContext context;
+
+    @Autowired
+    ObjectMapper objectMapper = new ObjectMapper();
+
+    @MockBean
+    private ProductService productService;
+
+    @BeforeEach
+    public void setUp() {
+        mockMvc = MockMvcBuilders.webAppContextSetup(context)
+                .apply(springSecurity())
+                .build();
+    }
+
+    @Test
+    public void addProductTest() throws Exception {
+        //given
+        ProductDTO dto = new ProductDTO(null, "pear", 33, BigDecimal.valueOf(35.89));
+        Product product = new Product(1000L, "pear", 33, BigDecimal.valueOf(35.89));
+
+        doReturn(product).when(productService).addProduct(eq(dto));
+
+        //when & then
+        mockMvc.perform(MockMvcRequestBuilders
+                        .post("/api/v1/products")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(objectMapper.writeValueAsString(dto)))
+                .andExpect(status().isCreated())
+                .andExpect(MockMvcResultMatchers.jsonPath("$.id", is(1000)))
+                .andExpect(MockMvcResultMatchers.jsonPath("$.title", is(product.getTitle())))
+                .andExpect(MockMvcResultMatchers.jsonPath("$.available", is(product.getAvailable())))
+                .andExpect(MockMvcResultMatchers.jsonPath("$.price").value((product.getPrice())));
+
+        verify(productService, times(1))
+                .addProduct(eq(dto));
+
+    }
+
+    @Test
+    public void getAllProductsTest() throws Exception {
+        Product product1 = new Product(1L, "pear", 33, BigDecimal.valueOf(35.89));
+        Product product2 = new Product(2L, "apple", 23, BigDecimal.valueOf(13.90));
+        List<Product> productList = new ArrayList<>();
```

Comment:
These 3 lines could be substituted with Arrays.asList(product1, product2);

Reply:
done